include/envoy/network/connection.h

```diff
@@ -240,13 +240,14 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
   virtual const ConnectionSocket::OptionsSharedPtr& socketOptions() const PURE;
 
   /**
-   * Object on which filters can share data on a per-connection basis.
-   * Only one filter can produce a named data object, but it may be
-   * consumed by many other objects.
-   * @return the per-connection state associated with this connection.
+   * The StreamInfo object associated with this connection. This is typically
+   * used for logging purposes. Individual filters may add specific information
+   * via the FilterState object within the StreamInfo object.
+   *
+   * @return StreamInfo object associated with this connection.
```

Comment:
nit: add a note here that the streamInfo object here is independent of the StreamInfo object inside HTTP connection manager which is per request, while the connection level one is per connection.